### ISSUE TITLE
commented node is considered as root node

### ISSUE THREAD

**Issue by msh-odoo** [odoo/owl#714]
Consider I have following template:
``` 
<t t-name="DrawerMenu">
    <!-- Some Comment -->
    <div>Hello World</div>
</t>
```

When I run component with such template, owl throws error, "A template should not have more than one root node".

I think if node is commented then owl should not consider it as root node and should not throw error.

**Comment by ged-odoo** [comment]
Hey @msh-odoo sorry for taking so long getting to this issue. I just tried using your template, and it seems to work. Could you try to reproduce your issue again?  And if it still happens, maybe give me some more details? Thanks!

**Comment by SimonGenin** [comment]
Seems to work, closing unless @msh-odoo comes with a counter example.

**Comment by msh-odoo** [comment]
@ged-odoo @SimonGenin Sorry, didn't get time to give you example, I will check again soon and will provide you example if I still face the issue, Thanks

**Comment by nla-odoo** [comment]
Hello @SimonGenin @ged-odoo 

I faced the same issue in the POS template.
when the debug mode is on, it will add one line at the root of the template like this, 
when I turn off the debug mode, it works as expected.
so I think it should allow developers to run it with debug mode on.

What is your view on this?

Thanks

**Comment by SimonGenin** [comment]
@nla-odoo That's interesting, we might have missed the bug because we weren't on debug mode. Your intervention narrows it down, will check again what could cause this.

**Comment by SimonGenin** [comment]
@nla-odoo Do you have a branch (and the line number of the file) where I could see the place the problem occurs ? So I can have more info on the context ?

**Comment by nla-odoo** [comment]
Hello @SimonGenin 
I did it in the Private repo of the project, But I can share the template that I created.

```
<t t-name='PaymentScreen' t-inherit="point_of_sale.PaymentScreen" t-inherit-mode="extension" owl="1">
    <xpath expr="//div[hasclass('payment-screen')]" position="replace">
        <div>Payment Body Here</div>
    </xpath>
</t>
```

For this template, I faced the issue.
I hope this can help you.

Thanks & Regards

**Comment by rb3606** [comment]
I have same issue. when in debug mode and extend pos template its giving error.

`<t t-name="OrderReceipt" t-inherit="point_of_sale.OrderReceipt" t-inherit-mode="extension" owl="1">
        <xpath expr="//div[hasclass('pos-receipt')]" position="replace">
</xpath>
    </t>`

**Comment by MahmoudFahmiEbs** [comment]
Hello @ged-odoo @SimonGenin is this still unhandled ?, cause i am facing same issue in this template :
```
<templates id="template" xml:space="preserve">
    <t t-name="CustomDiscount" t-inherit="pos_discount.DiscountButton" t-inherit-mode="extension" owl="1">
    <xpath expr="//span" position="replace">
         <span>
            <i class="fa fa-tag"></i>
            <span> </span>
            <span>Discount--</span>
        </span>
    </xpath>
    </t>
</templates> 
```

**Comment by SimonGenin** [comment]
@MahmoudFahmiEbs Do you also have to problem only in debug mode ?

**Comment by SimonGenin** [comment]
@MahmoudFahmiEbs The error is from your side, the xpath is invalid. 
If I do something like:
```
    <t t-name="CustomDiscount" t-inherit="pos_discount.DiscountButton" t-inherit-mode="extension" owl="1">
        <xpath expr="//span" position="replace">
                <span>Discount-</span>
        </xpath>
    </t>
```
That just works.

**Comment by MahmoudFahmiEbs** [comment]
@SimonGenin yeah it was jus an exampel , i want to replace the root node "span" first one with a button so i got this console log error "A template should not have more than one root node"
`<templates id="template" xml:space="preserve">
   <t t-name="CustomDiscount" t-inherit="pos_discount.DiscountButton" t-inherit-mode="extension" owl="1">
        <xpath expr="//span[hasclass('control-button')]" position="replace">

                <button class="control-button js_discount"
                t-att-class="{
                        'selected-mode': props.activeMode === 'discount',
                        'disabled-mode': !hasManualDiscount
                    }"
                    t-att-disabled="!hasManualDiscount">Discount-</button>
        </xpath>
    </t>
</templates>`

**Comment by MahmoudFahmiEbs** [comment]
<templates id="template" xml:space="preserve">
   <t t-name="CustomDiscount" t-inherit="pos_discount.DiscountButton" t-inherit-mode="extension" owl="1">
        <xpath expr="//span[hasclass('control-button')]" position="replace">

                <button class="control-button js_discount"
                t-att-class="{
                        'selected-mode': props.activeMode === 'discount',
                        'disabled-mode': !hasManualDiscount
                    }"
                    t-att-disabled="!hasManualDiscount">Discount-</button>
        </xpath>
    </t>
</templates>

**Comment by MahmoudFahmiEbs** [comment]
<templates id="template" xml:space="preserve">
   <t t-name="CustomDiscount" t-inherit="pos_discount.DiscountButton" t-inherit-mode="extension" owl="1">
        <xpath expr="//span[hasclass('control-button')]" position="replace">
                <button class="control-button js_discount"
                t-att-class="{
                        'selected-mode': props.activeMode === 'discount',
                        'disabled-mode': !hasManualDiscount
                    }"
                    t-att-disabled="!hasManualDiscount">Discount-</button>
        </xpath>
    </t>
</templates>

**Comment by MahmoudFahmiEbs** [comment]
@SimonGenin i can not paste code here , don know why plese find this 
![code2](https://user-images.githubusercontent.com/78018149/116680104-95a43300-a9ab-11eb-823c-d3ee7aa43d5f.png)

**Comment by SimonGenin** [comment]
@msh-odoo @nla-odoo @rb3606 @MahmoudFahmiEbs @ged-odoo 

Thank you all for your feedbacks. It was actually quite tedious to find. 
Basically, we discovered that Owl (1, version 2 will fix automatically many things) has a very hard time handling comments. And our playground tricked us as it stripped comments, making reproduction of the bug impossible. 

Here's a few things we discovered:

This complains as expected there is two node roots
```xml
<t t-name="App">
    hello 
    <div>test</div>
</t>
```

This returns (only) hello in the DOM, not what we would like
```xml
<t t-name="App"> 
    <div>test</div>
    hello
</t>
```

This returns the text "comment" in the dom, not what we would like
```xml
<t t-name="App"> 
    <!-- comment -->
</t>
```

This returns the text "comment" in the dom again, not what we would like
```xml
<t t-name="App"> 
   <div>Test</div>
    <!-- comment -->
</t>
```


This complains  for two root nodes. This is the problem you all had as it should just work properly.
```xml
<t t-name="App"> 
    <!-- comment -->
    <div>hello</div>
</t>
```

Now, when you did an xpath in odoo debug mode in extension mode, there was a comment added to keep track of which template modified which. Hence your error at the end. 

So, for now, we remove this comment in master https://github.com/odoo/odoo/pull/70227. 
This is not the end solution of course. But our current focus is on the webclient and Owl 2 will fix that so we don't spend to much time on that. 

Thank you !
